test/tests/api/v1_1/workflows_tests.py

```diff
@@ -117,26 +119,54 @@ def test_workflows_library_identifier_get(self):
         """ Testing GET:/library:/identifier"""
         Workflows().workflows_library_injectable_name_get(self.workflowDict.get('injectableName'))
         assert_equal(200,self.__client.last_response.status)
-        assert_equal(self.workflowDict.get('friendlyName'),str(json.loads(self.__client.last_response.data).get('friendlyName')))
+        assert_equal(self.workflowDict.get('friendlyName'), \
+            str(json.loads(self.__client.last_response.data).get('friendlyName')))
 
-    def post_workflows(self, graph_name):
-        Nodes().nodes_get()
-        nodes = loads(self.__client.last_response.data)
+    def post_workflows(self, graph_name, timeout_sec=300, nodes=[], data={}, callback=None):
         self.__graph_name = graph_name
-
-        for n in nodes:
-            if n.get('type') == 'compute':
-                id = n.get('id')
-                assert_not_equal(id,None)
-                LOG.info('starting amqp listener for node {0}'.format(id))
-                self.__task_worker=AMQPWorker(queue=QUEUE_GRAPH_FINISH,
-                                    callbacks=[self.handle_graph_finish])
+        self.__graph_status = []
+        
+        if len(nodes) == 0:
+            Nodes().nodes_get()
+            for n in loads(self.__client.last_response.data):
+                if n.get('type') == 'compute':
+                    nodes.append(n.get('id'))
+        
+        def thread_func(worker, id):
+            worker.start()
+            
+        if callback == None:
+            callback = self.handle_graph_finish
+        
+        for node in nodes:
+            LOG.info('Starting AMQP listener for node {0}'.format(node))
+            worker = AMQPWorker(queue=QUEUE_GRAPH_FINISH, callbacks=[callback])
+            self.__tasks.append(WorkerThread(worker, node)) 
+            retry = 10
+            while retry != 0:                   
                 try:
-                    Nodes().nodes_identifier_workflows_active_delete(id)
-                except Exception,e:
-                    assert_equal(404,e.status, message = 'status should be 404')
-                Nodes().nodes_identifier_workflows_post(id,name=graph_name,body={})
-                self.__task_worker.start()
+                    Nodes().nodes_identifier_workflows_active_delete(node)
+                    Nodes().nodes_identifier_workflows_active_get(node)
+                    if self.__client.last_response.status != 204:
+                        retry -= 1
+                        time.sleep(1)
+                    else:
+                        break
+                except Exception, e:
+                    assert_equal(404, e.status, message = 'Expected 404 response status')
+                    break

```

Comment:
From my test, after nodes_identifier_workflows_active_delete(), no matter it returns 404 or 204, the API follows it will  be retired, even if the actual workflow has been started at the first call:
```
2016-05-19 22:37:26,642 WARNING Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', BadStatusLine("''",))': /api/1.1/nodes/573d9acc1525e6593247fb79/workflows?name=Graph.PXE.Reboot
2016-05-19 22:37:26,642:urllib3.connectionpool:WARNING - Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', BadStatusLine("''",))': /api/1.1/nodes/573d9acc1525e6593247fb79/workflows?name=Graph.PXE.Reboot
```
From
```
curl http://localhost:8080/api/common/workflows/*
```
We can see that two same workflows are started. It might be caused by the untimely release of network resources when it is not a GET operation.

I tested that the API after nodes_identifier_workflows_active_get() won't be retried, so I add this command to shield the retry problem after deleting active workflow.
https://github.com/RackHD/RackHD/pull/232/files#diff-d618aea7eb216b7b27d43c843a01227cR153

You may consider whether it can be adopted.